src/libraries/System.Private.CoreLib/src/System/SearchValues/Strings/Helpers/RabinKarp.cs

```diff
@@ -64,6 +64,12 @@ public RabinKarp(ReadOnlySpan<string> values)
 
             foreach (string value in values)
             {
+                if (value.Length > MaxInputLength)
+                {
+                    // This value can never match. There's no point in including it in the buckets.
+                    continue;
```

Comment:
I'm not fully understanding why this can never match. This impl is only every user as part of other impls that are guaranteed to handle the larger values? Do we have tests that cover these cases?

Reply:
> This impl is only every user as part of other impls that are guaranteed to handle the larger values?

Yes. We'll only call into Rabin-Karp as a fallback for short inputs that Teddy can't handle.

There are a few more comments about this throughout this file, e.g.:
https://github.com/dotnet/runtime/blob/45ca92455760c7b4ad032d1d6868052178d5d46f/src/libraries/System.Private.CoreLib/src/System/SearchValues/Strings/Helpers/RabinKarp.cs#L13-L14

https://github.com/dotnet/runtime/blob/45ca92455760c7b4ad032d1d6868052178d5d46f/src/libraries/System.Private.CoreLib/src/System/SearchValues/Strings/Helpers/RabinKarp.cs#L28-L31

https://github.com/dotnet/runtime/blob/45ca92455760c7b4ad032d1d6868052178d5d46f/src/libraries/System.Private.CoreLib/src/System/SearchValues/Strings/Helpers/RabinKarp.cs#L55-L61

> Do we have tests that cover these cases?

We do through the `*RandomInputs*`/stress tests
https://github.com/dotnet/runtime/blob/79dd9bae9bb881eb716b608577c4cedc6c9cba72/src/libraries/System.Memory/tests/Span/StringSearchValues.cs#L248

In general we have very few tests manually exercising different paths. E.g. this may test the rabin-karp impl, but not Teddy since the inputs are too short.

Reply:
> In general we have very few tests manually exercising different paths

Could we add some? A random inputs test as a backup is great, but I'd feel more confident with an explicit set of tests covering at least some known border cases, corner cases, etc. It can be separate from this PR.